main/src/main/scala/sbt/Script.scala

```diff
@@ -32,7 +42,7 @@ object Script {
       val append = Load.transformSettings(Load.projectScope(currentRef), currentRef.build, rootProject, scriptSettings ++ embeddedSettings)
 
       val newStructure = Load.reapply(session.original ++ append, structure)
-      val arguments = state.remainingCommands.drop(1)
+      val arguments = state.remainingCommands.drop(1).map(arg => "\"" + arg + "\"")
```

Comment:
No, prior to that change, arguments with quotes and spaces did not work.
The quotes get stripped.
On Apr 11, 2016 7:23 PM, "eugene yokota" <notifications@github.com> wrote:

In main/src/main/scala/sbt/Script.scala
<https://github.com/sbt/sbt/pull/2549#discussion_r59313048>:

> @@ -32,7 +42,7 @@ object Script {
>        val append = Load.transformSettings(Load.projectScope(currentRef), currentRef.build, rootProject, scriptSettings ++ embeddedSettings)
>
>        val newStructure = Load.reapply(session.original ++ append, structure)
> -      val arguments = state.remainingCommands.drop(1)
> +      val arguments = state.remainingCommands.drop(1).map(arg => "\"" + arg + "\"")

What does this quoting do? Wouldn't this interfere if the existing users
are already adding quotes?

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub
<https://github.com/sbt/sbt/pull/2549/files/9b11ec2edc87e81c65a48f863a4b54757f73a125#r59313048>